### PR TITLE
[FrameworkBundle] Support use of hyphen in assets package name 

### PR DESCRIPTION
**Pull request by damaya**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes (Manual tests only)
| Fixed tickets | #28122
| License       | MIT
| Doc PR        | n/a

According to issue symfony/symfony-docs#10442, we tested in a demo bundle, for example in src/AppBundle/Resources/config/config.yml a package using hyphens: app-client-frontend, and withouth the patch it fails because the package is not recognized. With the patch, it works as expected.

```yaml
framework:
    assets:
        packages:
            app-client-frontend:
                version: "%env(FRONTEND_VERSION)%"
                version_format: '%%2$s/dist/%%1$s'
                base_urls:
                  - "%env(FRONTEND_URL)%"
```

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -631,7 +631,7 @@ private function addAssetsSection(ArrayNodeDefinition $rootNode)
                     ->end()
                     ->fixXmlConfig('package')
                     ->children()
-                        ->arrayNode('packages')
+                        ->arrayNode('packages')->normalizeKeys(false)
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: should be put on a separate line

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]
can you please enhance the commit message, and update the PR description according to the required template (filled in)?
see https://github.com/symfony/symfony/blob/master/.github/PULL_REQUEST_TEMPLATE.md

**Comment by fabpot** [comment]
Some test should be added as well (example from the related issue would work well here).

**Comment by damaya** [comment]
Sure, sorry the delay in my answer, I'll update the PR with required data.

**Review by damaya — COMMENTED** [review]
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes (Manual tests only)
| Fixed tickets | #28122
| License       | MIT
| Doc PR        | n/a

According to issue symfony/symfony-docs#10442, we tested in a demo bundle, for example in src/AppBundle/Resources/config/config.yml a package using hyphens: app-client-frontend, and withouth the patch it fails because the package is not recognized. With the patch, it works as expected.

```yaml
framework:
    assets:
        packages:
            app-client-frontend:
                version: "%env(FRONTEND_VERSION)%"
                version_format: '%%2$s/dist/%%1$s'
                base_urls:
                  - "%env(FRONTEND_URL)%"
```

**Comment by damaya** [comment]
Hi there, I updated my commit message and use a new line to improve the syntax. I just did manual tests and this patch worked as expected.
Thanks in advance.

**Comment by fabpot** [comment]
@damaya I still think that we need to add a test to avoid any regressions in the future. Can you do that or do you need help?

**Comment by damaya** [comment]
> @damaya I still think that we need to add a test to avoid any regressions in the future. Can you do that or do you need help?

I would love to do that, could you please give me a reference for how to implement tests for this according to the standard? thanks so much.

**Comment by XuruDragon** [comment]
The continuity of this PR has been achieved here : #30007

**Comment by nicolas-grekas** [comment]
Continued in #30007, thank you @damaya

**Pull request closed without merging**